diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9814380d..85d447ac8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,6 +51,7 @@ option(USE_SHARED_CURL    "Use your installed copy of curl" off)
 option(USE_SHARED_GIFLIB  "Use your installed copy of giflib" off)
 option(USE_SHARED_JPEGLIB "Use your installed copy of jpeglib" off)
 option(USE_SHARED_ZLIB    "Use your installed copy of zlib" off)
+option(USE_SHARED_LIBARCHIVE "Use your installed copy of libarchive" off)
 option(USE_SHARED_LIBPNG  "Use your installed copy of libpng" off)
 option(USE_SHARED_TINYXML "Use your installed copy of tinyxml" off)
 option(USE_SHARED_PIXMAN  "Use your installed copy of pixman" off)
@@ -210,6 +211,16 @@ else()
 endif()
 include_directories(${ZLIB_INCLUDE_DIRS})
 
+# libarchive
+if(USE_SHARED_LIBARCHIVE)
+  find_package(LibArchive REQUIRED)
+else()
+  set(LibArchive_FOUND)
+  set(LibArchive_LIBRARIES archive_static)
+  set(LibArchive_INCLUDE_DIRS $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libarchive/libarchive>)
+endif()
+include_directories(${LibArchive_INCLUDE_DIRS})
+
 # libpng
 if(USE_SHARED_LIBPNG)
   find_package(PNG REQUIRED)
